### PR TITLE
feat(ui) refactor notebook menu

### PR DESCRIPTION
**Pull request by wg102**
Closes https://github.com/StatCan/aaw/issues/1774

### CODE REVIEW

**Review by Jose-Matsuda — COMMENTED** [review]
everything seems to work fine, will let Mathis take his hand at it (using c56241ce5f73dbb2c875585768c08186cd81122d on dev)

File: frontend/jupyter/yarn.lock [hunk 1]
```diff
@@ -6269,12 +6269,6 @@
   "resolved" "https://registry.npmjs.org/klona/-/klona-2.0.6.tgz"
   "version" "2.0.6"
 
-"kubeflow@file:../common/kubeflow-common-lib/dist/kubeflow":
```
Comment (mathis-marcotte) [inline frontend/jupyter/yarn.lock:6272]: I don't know how much this actually would affect us, but it's probably best to NOT commit this change.

File: notebooks.go [hunk 2]
```diff
@@ -533,6 +534,11 @@ func (s *server) NewNotebook(w http.ResponseWriter, r *http.Request) {
 		notebook.Spec.Template.Spec.Containers[0].Resources.Limits[corev1.ResourceMemory] = req.MemoryLimit
 	}
 
+	// There's more than just the json
```
Comment (mathis-marcotte) [inline notebooks.go:537]: What do you mean that there's more than just the json?

File: frontend/jupyter/src/app/pages/form/form-new/form-new.component.html [hunk 3]
```diff
@@ -29,41 +29,67 @@
...
-      >
-      </app-form-workspace-volume>
-
-      <app-form-data-volumes
-        [volsArray]="formCtrl.get('datavols')"
-        [readonly]="config?.dataVolumes?.readOnly"
-        [externalName]="formCtrl.get('name').value"
-      >
-      </app-form-data-volumes>
+      <lib-form-section title="Protected B">
```
Comment (mathis-marcotte) [inline frontend/jupyter/src/app/pages/form/form-new/form-new.component.html:32]: I feel like we could probably convert this into its own component (like with what you did with the form-image-custom)

File: frontend/jupyter/src/app/pages/form/form-new/form-image-custom/form-image-custom.component.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+import { V1Namespace } from '@kubernetes/client-node';
+import { Config } from 'src/app/types';
+import { MatCheckboxChange } from '@angular/material/checkbox';
+
+@Component({
+  selector: 'app-form-image-custom',
+  templateUrl: './form-image-custom.component.html',
+  styleUrls: ['./form-image-custom.component.scss'],
+})
+export class FormImageCustomComponent implements OnInit, OnDestroy {
```
Comment (mathis-marcotte) [inline frontend/jupyter/src/app/pages/form/form-new/form-image-custom/form-image-custom.component.ts:28]: I think you should re-evaluate what needs to be in this component. From what I can tell, this was created from a copy paste of the form-image component, and there's a lot of lingering code from that component that is not needed in here.

**Review by mathis-marcotte — CHANGES_REQUESTED** [review]

File: notebooks.go (hunk 2 above)
Comment (wg102) [reply notebooks.go:537]: This is the wrong comment. Fixed

**Review by mathis-marcotte — APPROVED** [review]
Changes requested were done and things look good to me.

**Pull request merged**